### ISSUE TITLE
[DatePicker](Community): textField InputProps endAdornment cannot replace icon

### ISSUE THREAD

**Issue by devChrisho** [mui/mui-x#9899]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Steps:
1. Add DesktopDatePicker 
```   
    <DesktopDatePicker
        slotProps={{
          textField: {
            InputProps: {
              endAdornment: <EmojiEmotionsIcon />,
            },
          },
        }}
      />
```
3. Compare against MobileDatePicker
```
      <MobileDatePicker
        slotProps={{
          textField: {
            InputProps: {
              endAdornment: <EmojiEmotionsIcon />,
            },
          },
        }}
      />
```
4. Compare against DatePicker (small device)
```
      <DatePicker
        slotProps={{
          textField: {
            InputProps: {
              disabled: true,
              endAdornment: <EmojiEmotionsIcon />,
            },
          },
        }}
      />
```

### Current behavior 😯

<img width="370" alt="Screenshot 2023-08-03 at 5 37 54 PM" src="https://github.com/mui/mui-x/assets/68413782/5bf3147b-cf70-4b5f-9a4a-c3f58b9a72ba">

![code](https://github.com/mui/mui-x/assets/68413782/1284454f-7fc1-4c88-8730-9a256cd7dd4b)

### Expected behavior 🤔

The endAdorment icon should be changed. 

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: macOS 13.4.1
  Binaries:
    Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
  Browsers:
    Chrome: 115.0.5790.114
    Edge: Not Found
    Safari: 16.5.2
  npmPackages:
    @emotion/react: ^11.11.1 => 11.11.1 
    @emotion/styled: ^11.11.0 => 11.11.0 
    @mui/base:  5.0.0-beta.8 
    @mui/core-downloads-tracker:  5.14.3 
    @mui/icons-material: ^5.14.3 => 5.14.3 
    @mui/lab: ^5.0.0-alpha.137 => 5.0.0-alpha.137 
    @mui/material: ^5.14.3 => 5.14.3 
    @mui/private-theming:  5.13.7 
    @mui/styled-engine:  5.13.2 
    @mui/styles: ^5.14.3 => 5.14.3 
    @mui/system:  5.14.3 
    @mui/types:  7.2.4 
    @mui/utils:  5.14.3 
    @mui/x-date-pickers: ^6.10.2 => 6.10.2 
    @types/react: ^17.0.44 => 17.0.44 
    react: ^18.1.0 => 18.1.0 
    react-dom: ^18.1.0 => 18.1.0 
    typescript: ^4.3.2 => 4.3.2 

browser: chrome
```
</details>

**Comment by flaviendelangle** [comment]
Hi,
Thanks for your contribution.

Could you provide a working reproduction, this [Codesandbox template](https://codesandbox.io/s/date-and-time-pickers-uj6btq) might be a good starting point.

**Comment by LukasTy** [comment]
Closing as a duplicate of https://github.com/mui/mui-x/issues/9844.
It has already been resolved and released in `6.11.0`. Please update your package to receive the fix.

FYI: If you are replacing the whole `endAdornment` with a simple icon, you will lose the open picker functionality on the desktop variant.
Consider defining the icon in the following way instead:
```tsx
slots={{
  openPickerIcon: EmojiEmotionsIcon,
}}
```
But do note, that this will not work on the mobile variant as it currently does not have such a slot.
